### PR TITLE
docs(http): add docs and examples for HTTP_BINDINGS

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

File: modules/angular2/http.ts [hunk 1]
```diff
@@ -47,19 +47,117 @@ export {URLSearchParams} from './src/http/url_search_params';
...
+ * import {bind, Injector} from 'angular2/angular2';
+ * import {HTTP_BINDINGS, Http, Response, XHRBackend, MockBackend} from 'angular2/http';
+ * import {ObservableWrapper} from 'angular2/src/core/facade/async';
+
+ * var people = [{name: 'Jeff'}, {name: 'Tobias'}];
+
+ * var injector = Injector.resolveAndCreate([
+ *   HTTP_BINDINGS,
+ *   MockBackend,
+ *   bind(XHRBackend).toAlias(MockBackend)
```
Comment (PascalPrecht) [inline modules/angular2/http.ts]: Couldn't we also just `bind(XHRBackend).toClass(MockBackend)` here?

File: modules/angular2/http.ts [hunk 2]
```diff
@@ -47,19 +47,116 @@ export {URLSearchParams} from './src/http/url_search_params';
...
+ *         <li *ng-for="#person of people">
+ *           {{person.name}}
+ *         </li>
+ *       </ul>
+ *     </div>
+ *   `,
+ *   directives: [NgFor]
+ * })
+ * export class App {
+ *   people: Array<Object>;
```
Comment (gkalpak) [inline modules/angular2/http.ts]: Isn't the `Object[]` form to be preferred (over `Array<Object>`) ?

File: modules/angular2/http.ts [hunk 3]
```diff
@@ -47,19 +47,116 @@ export {URLSearchParams} from './src/http/url_search_params';
...
  *   constructor(http:Http) {
- *     http.request('data.txt').subscribe(res => this.data = res.text());
+ *     http.get('people.json').toRx().subscribe(res => {
+ *       this.people = res.json();
+ *     })
```
Comment (gkalpak) [inline modules/angular2/http.ts]: ;

File: modules/angular2/http.ts [hunk 4]
```diff
@@ -47,19 +47,116 @@ export {URLSearchParams} from './src/http/url_search_params';
...
+ * There may be cases where it makes sense to extend the base request options,
+ * such as to add a search string to be appended to all URLs.
+ * To accomplish this, a new binding for {@link RequestOptions} should
+ * be added in the same injector as `HTTP_BINDINGS`.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/aCMEXi?p=preview))
+ *
+ * ```
+ * import {bind, bootstrap} from 'angular2/angular2';
+ * import {HTTP_BINDINGS, Http, BaseRequestOptions, RequestOptions} from 'angular2/http';
```
Comment (gkalpak) [inline modules/angular2/http.ts]: Is `Http` necessary here ?

File: modules/angular2/http.ts [hunk 5]
```diff
@@ -47,19 +47,116 @@ export {URLSearchParams} from './src/http/url_search_params';
...
+ * binding should be bound to {@link MockBackend}.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/7LWALD?p=preview))
+ *
+ * ```
+ * import {bind, Injector} from 'angular2/angular2';
+ * import {HTTP_BINDINGS, Http, Response, XHRBackend, MockBackend} from 'angular2/http';
+ * import {ObservableWrapper} from 'angular2/src/core/facade/async';
+
+ * var people = [{name: 'Jeff'}, {name: 'Tobias'}];
```
Comment (gkalpak) [inline modules/angular2/http.ts]: `people` isn't used below.

File: modules/angular2/src/http/headers.ts [hunk 6]
```diff
@@ -10,8 +10,29 @@ import {
...
- * difference from the spec is the lack of an `entries` method.
+ * specified in the [Fetch Spec](https://fetch.spec.whatwg.org/#headers-class).
+ *
+ * The only known difference between this `Headers` implementation and the spec is the
+ * lack of an `entries` method.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/MTdwT6?p=preview))
+ *
+ * ```
+ * import {Headers} from "angular2/http";
```
Comment (gkalpak) [inline modules/angular2/src/http/headers.ts]: <sub>(Single quotes for consistency ?)</sub>

File: modules/angular2/http.ts [hunk 7]
```diff
@@ -73,6 +170,112 @@ export const HTTP_BINDINGS: any[] = [
...
+ * There may be cases where it makes sense to extend the base request options,
+ * such as to add a search string to be appended to all URLs.
+ * To accomplish this, a new binding for {@link RequestOptions} should
+ * be added in the same injector as `JSONP_BINDINGS`.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/TFug7x?p=preview))
+ *
+ * ```
+ * import {bind, bootstrap, Component, NgFor, View} from 'angular2/angular2';
+ * import {JSONP_BINDINGS, Jsonp, BaseRequestOptions, RequestOptions} from 'angular2/http';
```
Comment (gkalpak) [inline modules/angular2/http.ts]: `Component`, `NgFor`, `View` and `Jsonp` aren't used.

File: modules/angular2/http.ts [hunk 8]
```diff
@@ -73,6 +170,112 @@ export const HTTP_BINDINGS: any[] = [
...
+ * }
+ *
+ * bootstrap(App, [JSONP_BINDINGS, bind(RequestOptions).toClass(MyOptions)])
+ *   .catch(err => console.error(err));
+ * ```
+ *
+ * Likewise, to use a mock backend for unit tests, the {@link JSONPBackend}
+ * binding should be bound to {@link MockBackend}.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/HDqZWL?p%3Dpreview&p=preview))
```
Comment (gkalpak) [inline modules/angular2/http.ts]: The URL seems weird (but works). This one works just as well (and looks less weird): http://plnkr.co/edit/HDqZWL?p=preview

File: modules/angular2/http.ts [hunk 9]
```diff
@@ -47,19 +47,116 @@ export {URLSearchParams} from './src/http/url_search_params';
...
  *
+ * Likewise, to use a mock backend for unit tests, the {@link XHRBackend}
+ * binding should be bound to {@link MockBackend}.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/7LWALD?p=preview))
+ *
+ * ```
+ * import {bind, Injector} from 'angular2/angular2';
+ * import {HTTP_BINDINGS, Http, Response, XHRBackend, MockBackend} from 'angular2/http';
+ * import {ObservableWrapper} from 'angular2/src/core/facade/async';
```
Comment (gkalpak) [inline modules/angular2/http.ts]: Is `ObservableWrapper` needed ?

File: modules/angular2/http.ts [hunk 10]
```diff
@@ -73,6 +170,112 @@ export const HTTP_BINDINGS: any[] = [
...
+ * Likewise, to use a mock backend for unit tests, the {@link JSONPBackend}
+ * binding should be bound to {@link MockBackend}.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/HDqZWL?p%3Dpreview&p=preview))
+ *
+ * ```
+ * import {bind, Injector} from 'angular2/angular2';
+ * import {JSONP_BINDINGS, Jsonp, Response, JSONPBackend, MockBackend} from 'angular2/http';
+ *
+ * var people = [{name: 'Jeff'}, {name: 'Tobias'}];
```
Comment (gkalpak) [inline modules/angular2/http.ts]: `people` isn't used below.

File: modules/angular2/src/http/static_request.ts [hunk 11]
```diff
@@ -18,19 +18,45 @@ import {
  * Spec](https://fetch.spec.whatwg.org/#request-class),
  * but is considered a static value whose body can be accessed many times. There are other
  * differences in the implementation, but this is the most significant.
+ *
+ * `Request` instances are typically created by higher-level classes, like {@link Http} and
+ * {@link Jsonp}, but it may be occasionally be useful to explicitly create `Request` instances.
```
Comment (vsavkin) [inline modules/angular2/src/http/static_request.ts]: `it may occasionally be useful`

File: modules/angular2/http.ts (hunk 10 above)
Comment (jeffbcross) [inline modules/angular2/http.ts]: @gkalpak thanks for all the feedback, I've made all changes locally (and on plunker).

File: modules/angular2/http.ts [hunk 12]
```diff
@@ -47,19 +47,115 @@ export {URLSearchParams} from './src/http/url_search_params';
...
+ * })
+ * export class App {
+ *   people: Object[];
  *   constructor(http:Http) {
- *     http.request('data.txt').subscribe(res => this.data = res.text());
+ *     http.get('people.json').toRx().subscribe(res => {
+ *       this.people = res.json();
+ *     });
  *   }
+ *   active:boolean = false;
```
Comment (todoubaba) [inline modules/angular2/http.ts]: `active` is not used in the component.

Comment (mgechev) [inline modules/angular2/http.ts]: It is used in `toggleActiveState`.

File: modules/angular2/src/http/static_request.ts [hunk 13]
```diff
@@ -18,19 +18,45 @@ import {
...
+ * ```
+ * import {Injectable, Injector} from 'angular2/angular2';
+ * import {HTTP_BINDINGS, Http, Request} from 'angular2/http';
+ *
+ * @Injectable()
+ * class AutoAuthenticator {
+ *   constructor(public http:Http) {}
+ *   request(url:string) {
+ *     return this.http.request(new Request({
+ *       method: 0, //GET.
```
Comment (mgechev) [inline modules/angular2/src/http/static_request.ts]: Why not `RequestMethds.Get` instead? This is TypeScript syntax anyway.

File: modules/angular2/http.ts (hunk 12 above)
Comment (todoubaba) [inline modules/angular2/http.ts]: `toggleActiveState` is also not used in the template or Http related code. They are nothing to do with the http example.

**Pull request closed without merging**